R/stats-mlm-tidiers.R

```diff
@@ -0,0 +1,99 @@
+#' @templateVar class mlm
+#' @template title_desc_tidy
+#'
+#' @param x An `mlm` object created by [stats::lm()] with a matrix as the
+#'   response.
+#' @template param_confint
+#' @template param_quick
+#' @template param_unused_dots
+#'
+#' @evalRd return_tidy(regression = TRUE)
+#'
+#' @details In contrast to `lm` object (simple linear model), tidy output for
+#'   `mlm` (multiple linear model) objects contain an additional column
+#'   `response`.
+#'
+#'   If you have missing values in your model data, you may need to refit
+#'   the model with `na.action = na.exclude`.
+#'
+#' @examples
+#'
+#' mod <- lm(cbind(mpg, disp) ~ wt, mtcars)
+#' tidy(mod, conf.int = TRUE)
+#' @export
+#' @seealso [tidy()]
+#' @family lm tidiers
+#' @export
+
+tidy.mlm <- function(x,
+                     conf.int = FALSE,
+                     conf.level = .95,
+                     ...) {
+
+  # adding other details from summary object
+  s <- summary(x)
+  ret <- tidy.summary.mlm(s)
+
+  # adding confidence intervals
+  if (isTRUE(conf.int)) {
+
+    # S3 method for computing confidence intervals for `mlm` objects was
+    # introduced in R 3.5
+    CI <- tryCatch(
+      expr = stats::confint(x, level = conf.level),
+      error = function(x) {
+        NULL
+      }
+    )
+
+    # if R version is prior to 3.5, use the custom function
+    if (is.null(CI)) {
+      CI <- confint.mlm(x, level = conf.level)
+    }
+
+    colnames(CI) <- c("conf.low", "conf.high")
+    ret <- cbind(ret, unrowname(CI))
+  }
+
+  tibble::as_tibble(ret)
```

Comment:
`as_tibble()` is imported so no need to namespace it.

```suggestion
  as_tibble(ret)
```